bin/upload_new_release_to_s3.py

```diff
@@ -0,0 +1,47 @@
+import boto3
+import os
+import datetime
+import argparse
+
+parser = argparse.ArgumentParser(description='Usage')
+parser.add_argument('-r','--release-name', type=str,
+                    help="Name of release")
+args = parser.parse_args()
+
+if not args.release_name:
+  raise Exception("Please include a name for your release")
```

Comment:
If we're planning to switch to an automatic CICD release on `aitt_release_data` soon it's probably nbd, but I think a standardized release name with a date would be better than manually naming the release

Reply:
Agreed. I believe what Dave and I had discussed was using the branch name and date to name the release, which shouldn't be too hard to implement. I'm going to chat with infra about getting that set up sooner than later.

Reply:
it sounds like we've had a few pairwise conversations on this and I think I might be out of the loop on some discussion too. I was thinking that it would be ideal if we could use the release name to quickly find the corresponding contents in github, and the first thing that comes to mind is to have the git branch name match the release name. What do you think about:
1. have the aitt_release_data branch be in the format of YYYY-MM-DD-your-description-here
2. have this script verify the aitt branch is in that format, and then use it as the release name

side note, I made sure that "auto delete branches for merged PRs" is disabled in the aitt_release_data repo, and if we like this strategy we should try to remember not to delete branches after merging PRs in that repo.

Reply:
> I think a standardized release name with a date would be better than manually naming the release

not sure if my comments are agreeing or disagreeing with this statement Ceara 😅 I'd also be fine with just using a datestamp if that sounds better to you, though in that case I think we need somewhere in the aitt repo (a release notes file?) that we could look to see a name / description for what changed in that release.

Reply:
I think we're in agreement generally (release names should have a name that makes it easy to see what the latest is and it should be easy to find the corresponding contents in git) and we are figuring out the easiest way to do so

Reply:
IMO we should `git tag` each merge to main in `aitt_release_data` with either the date or major.minor.patch and make a tag release w/ CICD to s3. Then you could just checkout the tag in git to get the content

Reply:
i.e. we should do this: https://github.com/code-dot-org/code-dot-org/releases

Reply:
💯 on tags! Mark, do you want to choose between making the branch name match the release name, or switching this to use tags instead? the tags change initially seemed like a slightly bigger amount of effort that I didn't want to block the claude move, but (now that Karim is unblocked via adhoc) it seems reasonable to do as follow up if there is sufficient motivation

Reply:
My vote would be tags or commit hashes. Tags make a lot of sense, given that these are indeed releases.

Reply:
In the interest of resolving this conversation - is name good for now or does this need to change to advise something more specific (like version number or date)

Reply:
making the branch name match the date-stamped release name also works fine for me if that's any easier, and we'll just keep in mind that before a certain date, releases were branches and not tags.

Reply:
> let's keep our releases using the YYYY-MM-DD- prefix for now, which rules out using commit hashes as the release names. let's plan on tags as the long-term solution.
> 
> knowing that, how about once this is merged, you manually make a tag which matches the name of the date-prefixed release? this would support the goal of making it easy to locate the release contents in github given the release name, while building in the right direction.

@Nokondi can you confirm if this sounds good?